stablehlo/dialect/TypeInference.cpp

```diff
@@ -2390,11 +2390,26 @@ LogicalResult inferSliceOp(Optional<Location> location, Value operand,
   return success();
 }
 
-LogicalResult inferSortOp(
-    Optional<Location>, ValueRange inputs,
-    SmallVectorImpl<ShapedTypeComponents>& inferredReturnShapes) {
-  for (auto resultType : inputs.getTypes())
-    inferredReturnShapes.emplace_back(resultType.cast<ShapedType>());
+LogicalResult inferSortOp(Optional<Location> location, ValueRange inputs,
+                          SmallVectorImpl<Type>& inferredReturnTypes) {
+  // Infermost specific type from all operands
+  SmallVector<Type> mostSpecificType;
+  if (failed(hlo::inferMostSpecificType(location, inputs.getTypes(),
+                                        mostSpecificType))) {
```

Comment:
Sounds like we would also want an overload for `inferMostSpecificType` which takes `Type&`.